controllers/ControllerHome.php

```diff
@@ -2,41 +2,33 @@
 
 namespace OC_Blog\Controllers;
 
-use GuzzleHttp\Psr7\ServerRequest;
-use OC_Blog\Config\ConstantGlobal;
 use OC_Blog\Models\PostsManager;
+use OC_Blog\Tools\ControllerFactory;
 use OC_Blog\Tools\Session;
 
 
-class ControllerHome {
+class ControllerHome extends ControllerFactory {
 
-	private object $_twig;
-	private string $_server;
-
-	public function __construct( $twig ) {
-
-		$this->_twig = $twig;
-		$this->_server = ( new ConstantGlobal(ServerRequest::fromGlobals()) )->getServerName()['SERVER_NAME'];
-
-		$this->renderHome();
-
-	}
-
-	public function renderHome(){
+	/**
+	 * Affiche la page d'accueil.
+	 */
+	public function home(): void {
 		$posts = new PostsManager();
 		$allPosts = $posts->listPosts();
 		$key = (new Session)->getKey('user');
 
+
 		if(!empty($key)){
-			echo $this->_twig->render( 'home.twig', ['allPosts' => $allPosts,
+			echo $this->getTwig()->render( 'home.twig', ['allPosts' => $allPosts,
```

Comment:
Codacy found an issue: [All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$this'.](https://app.codacy.com/manual/PatOpen/OC_Blog/pullRequest?prid=6061935)